pkg/volume/csi/csi_mounter.go

```diff
@@ -217,12 +201,6 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		return fmt.Errorf("volume source not found in volume.Spec")
 	}
 
-	// create target_dir before call to NodePublish
```

Comment:
Thank you for the patch, @andyzhangx. I noticed that we used to call `os.MkdirAll` here, but now we won't. Can you clarify whether the CSI plugin should still find that this target directory already exists, due to it being created earlier, or are we transferring new responsibility to drivers to create this target directory?

Reply:
I tend to create this target directory by driver itself, e.g. on windows, target directory is actually a symbolic link, if we create target directory here, then on Windows CSI driver, it needs to remove this target directory and create a symbolic link.
cc @kkmsft @ddebroy

Reply:
I see. In my driver, and those that I've studied, we create the target path in the `NodePublishVolume` handler if the directory doesn't already exist, so I wouldn't mind this change in behavior.

Reply:
CSI says:

>   // The CO SHALL ensure that the parent directory of this path exists
  // and that the process serving the request has `read` and `write`
  // permissions to that parent directory.
  ...
  // Creation of target_path is the responsibility of the SP.

Should kubelet `MkdirAll()` the parent then?

Reply:
fixed

Reply:
I meant, creation of `deviceMountPath` is responsiblity of a CSI driver, while kubelet should create just the parent, i.e. `MkdirAll(Dir(deviceMountPath))`.

Note that the change could break some CSI drivers that expected kubelet to create whole path. I'm not sure anyone tested with something else than Kubernetes... @msau42, what do you think?

Reply:
@jsafrane o, that's what I did in the beginning, only create parent dir, and csi driver create `deviceMountPath`, and also I am not sure it would break sth. 
I reverted back the first commit and now this PR only checks the error when `MkdirAll(dir)`, it should be safer.

My initial opinion about this:
```
I tend to create this target directory by driver itself, e.g. on windows, target directory is actually a symbolic link, if we create target directory here, then on Windows CSI driver, it needs to remove this target directory and create a symbolic link.
```

Reply:
We have an issue for this already and yes, it's a problem for windows: https://github.com/kubernetes/kubernetes/issues/75535

I am ok with removing creation of mount path, but let's call it out in the release notes so that drivers that were uncompliant have a chance to fix it

Reply:
I had similar thought while getting the azuredisk csi driver to work in windows. But there are other drivers (eg. https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/pkg/secrets-store) which assumes the directory existance. Hence choose to work around this issue in the windows azure disk csi. My recommendation is that we hold off until 1.18 release rolls out. Make the change early in 1.19 and inform folks via sig storage channels - slack, mailing list , meetings about this giving opportunity to work the changes out by the time 1.19 comes out. Thoughts ?

Reply:
Sounds good. How about we at least make a deprecation notice this release?

Reply:
Are we mainly concerned with removal of mount path creation?

Do we want to proceed with removal of mount point check?

Reply:
> Sounds good. How about we at least make a deprecation notice this release?

Yes, SGTM 👍

Reply:
I think we should put in all three notes in deprecation notice 

1. Create if not present of the mount directory, 
2. Check whether its mounted 
3. Corruption check 

all getting deprecated in the release note and proceed with the removal at the beginning of the 1.19 cycle.  Although I don’t know of which drivers would break in the (2) and (3), it looks very easy possibility to break things late the cycle with this.